File: cm-maven-plugin/src/main/java/se/peterhenell/cm/parsers/JiraIssueParser.java
package se.peterhenell.cm.parsers;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

import se.peterhenell.cm.Logging;
import se.peterhenell.cm.dto.IssueInfo;

public class JiraIssueParser implements IssueParser {

	public JiraIssueParser() {
	}

	@Override
	public IssueInfo parse(String gitCommitMessage) {
		IssueInfo dto = IssueInfo.UnknownIssue;
		
		Pattern r = Pattern.compile("(.*)-(\\d+)(.*)");

		Matcher m = r.matcher(gitCommitMessage);
		if (m.find()) {
			if (m.groupCount() >= 3) {
				String projectKey = m.group(1).trim();
				String jiraIssue = m.group(2).trim();
				String message = m.group(3).trim();

				dto = IssueInfo.create(projectKey, jiraIssue, message);
			}
		}
		
		Logging.getLog().debug("Parse Result : " + dto.toString());
		return dto;
	}

}

File: cmmanagermojo/src/test/java/se/peterhenell/cmojo/JiraIssueParser.scala
package se.peterhenell.cmojo.parsers

import se.peterhenell.cmojo.dto.IssueInfo

/**
  * Created by mrm on 2017-06-06.
  */

class JiraIssueParser extends IssueParser {
  override def parse(gitCommitMessage: String): IssueInfo = {
    val r = """(.*)-(\d+)(.*)""".r

    gitCommitMessage match {
      case r(projectKey, jiraIssue, message) => IssueInfo.create(projectKey.trim, jiraIssue.trim, message.trim)
      case _ => IssueInfo.UnknownIssue
    }
  }
}
